frontend/src/pages/MetricCreateEdit.jsx

```diff
@@ -0,0 +1,181 @@
+import React, { useState, useEffect, useRef } from 'react'
+import { useParams, useNavigate } from 'react-router-dom';
+import TagsBox from '../components/TagsBox';
+import Layout from '../layouts/Layout';
+
+
+const MetricCreateEdit = (props) => {
+  const navigate = useNavigate();
+
+  const {metric_id} = useParams();
+  const [tags, setTags] = useState([]);
+  const [charLimit, setCharLimit] = useState(0);
+  
+  const metricNameRef = useRef(null);
+  const metricTypeRef = useRef(null);
+  const metricDescRef = useRef(null);
+    
+
+  const postMetric = async (e) => {
+    
+    const metricName = metricNameRef.current;
+    const metricType = metricTypeRef.current;
+    const metricDesc = metricDescRef.current; 
+
+    let formData = {
+        name: metricName.value,
+        type: metricType.value,
+        area: tags,
+        description: metricDesc.value
+    }
+
+    var data = await fetch('http://127.0.0.1:8000/api/v1/metrics', {
```

Comment:
it would be better if the fetch request and its handling is abstracted to a separate function called createMetric or something.
Also it would be nice if there's a folder called api where we create file for each api resource that defines the methods for this resource, in addition to the index.js file that exports a variable called BASE_URL and API_URL, where the BASE_URL will be set to the backend domain `http://127.0.0.1:8000` and the API_URL will be set to `/api/v1/`. the two variables can be configured later to take their values from env variables.

for example, api/metrics/index.ts  and there's a function called createMetric inside this file that takes the formData as a parameter and internally it uses fetch to send the request and returns the json response.

Reply:
the createMetric function goal is to send the request to the BE to create the metric, while the current postMetric function goal is to prepare the formData and call the createMetric passing the formData to it as a parameter.

you should perform validations on the data before calling the createMetric function, so you can define another function called validateMetricData and call it before preparing the data inside postMetric and if the validations fail terminate immediately and don't call the createMetric function.